samples/basic/clock_pwm/src/main.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2023 Andriy Gelman <andriy.gelman@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+
+#define NODELABEL DT_NODELABEL(samplenode)
+
+int main(void)
+{
+	const struct device *clk_dev = DEVICE_DT_GET(DT_CLOCKS_CTLR(NODELABEL));
+	clock_control_subsys_t id = (clock_control_subsys_t)DT_CLOCKS_CELL(NODELABEL, id);
+	int ret;
+	uint32_t rate;
+
+	printk("PWM-based clock control\n");
+	if (!device_is_ready(clk_dev)) {
+		printk("Error: PWM clock device %s is not ready\n", clk_dev->name);
+		return 0;
+	}
+
+	ret = clock_control_on(clk_dev, id);
+	if (ret) {
+		printk("Error starting clock %d\n", ret);
+		return ret;
+	}
+	printk("Started clock\n");
+
+	ret = clock_control_get_rate(clk_dev, id, &rate);
+	if (ret) {
+		printk("Error reading clock rate\n");
+		return ret;
+	}
+	printk("Clock frequency set to %dHz\n", rate);
+
+	k_sleep(K_SECONDS(10));
+	rate /= 2;
+	printk("New target clock frequency %dHz\n", rate);
+	ret = clock_control_set_rate(clk_dev, id, (clock_control_subsys_rate_t)rate);
+	if (ret) {
+		printk("Error reducing clock frequency %d\n", ret);
+		return ret;
+	}
+
+	ret = clock_control_get_rate(clk_dev, id, &rate);
+	if (ret) {
+		printk("Error reading clock rate\n");
+		return ret;
+	}
+	printk("Clock frequency set to %dHz\n", rate);
+	printk("Success\n");
```

Comment:
Main aim of a sample should be to demonstrate how to use a component in an application.
If the target is to perform a validation, add a test.
I'd suggest removing the "Success" here to avoid mixing purposes

Reply:
thanks, @erwango, I updated the PR.